src/CommunityToolkit.Maui.UnitTests/Layouts/StateLayoutTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Layouts;
+using CommunityToolkit.Maui.Views;
+using FluentAssertions;
+using Nito.AsyncEx;
+using Xunit;
+
+namespace CommunityToolkit.Maui.UnitTests.Layouts;
+
+public class StateLayoutTests : BaseTest
```

Comment:
This is a good start! Let's make sure we have a test that covers every public API in the new code: `StateLayout`, `StateLayoutController`, and `StateView`.

Reply:
Make sure you're also testing every edge case, eg What happens when `animationTokenSource` is canceled?

Reply:
Added a bunch more tests.